functions/Remove-DbaDatabaseSafely.ps1

```diff
@@ -177,205 +175,32 @@ function Remove-DbaDatabaseSafely {
-            $restore = New-Object 'Microsoft.SqlServer.Management.Smo.Restore'
-            $restore.ReplaceDatabase = $true
-
-            foreach ($file in $filestructure.values) {
-                $movefile = New-Object 'Microsoft.SqlServer.Management.Smo.RelocateFile'
-                $movefile.LogicalFileName = $file.logical
-                $movefile.PhysicalFileName = $file.physical
-                $null = $restore.RelocateFiles.Add($movefile)
-            }
-
-            try {
-                if ($TSql) {
-                    $restore.PercentCompleteNotification = 1
-                    $restore.add_Complete($complete)
-                    $restore.ReplaceDatabase = $true
-                    $restore.Database = $dbname
-                    $restore.Action = $filetype
-                    $restore.NoRecovery = $norecovery
-                    $device = New-Object -TypeName Microsoft.SqlServer.Management.Smo.BackupDeviceItem
-                    $device.name = $backupfile
-                    $device.devicetype = 'File'
-                    $restore.Devices.Add($device)
-                    $restorescript = $restore.script($server)
-                    return $restorescript
-                } else {
-                    $percent = [Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler] {
-                        Write-Progress -id 1 -activity "Restoring $dbname to $servername" -percentcomplete $_.Percent -status ([System.String]::Format("Progress: {0} %", $_.Percent))
-                    }
-                    $restore.add_PercentComplete($percent)
-                    $restore.PercentCompleteNotification = 1
-                    $restore.add_Complete($complete)
-                    $restore.ReplaceDatabase = $true
-                    $restore.Database = $dbname
-                    $restore.Action = $filetype
-                    $restore.NoRecovery = $norecovery
-                    $device = New-Object -TypeName Microsoft.SqlServer.Management.Smo.BackupDeviceItem
-                    $device.name = $backupfile
-                    $device.devicetype = 'File'
-                    $restore.Devices.Add($device)
-
-                    Write-Progress -id 1 -activity "Restoring $dbname to $servername" -percentcomplete 0 -status ([System.String]::Format("Progress: {0} %", 0))
-                    $restore.sqlrestore($server)
-                    Write-Progress -id 1 -activity "Restoring $dbname to $servername" -status 'Complete' -Completed
-
-                    return $true
-                }
-            } catch {
-                Stop-Function -Message "Restore failed" -ErrorRecord $_ -Target $dbname
-                return $false
-            }
-        }
-
     }
     process {
         if (Test-FunctionInterrupt) {
             return
         }
         try {
-            Start-SqlAgent
+            $destInstanceName = $destserver.InstanceName
+            if ($destserver.InstanceName -eq '') {
```

Comment:
This should include test for Edition because on Express this will cause the command to fail.

Reply:
Thanks! 
Done.